cvmfs/receiver/commit_processor.cc

```diff
@@ -250,33 +248,13 @@ CommitProcessor::Result CommitProcessor::Process(
                lease_path.c_str());
   }
 
-  {
-    UniquePtr<ServerTool> server_tool(new ServerTool());
-
-    if (!server_tool->InitDownloadManager(true, params.proxy)) {
-      LogCvmfs(
-          kLogReceiver, kLogSyslogErr,
-          "CommitProcessor - error: Could not initialize the download manager");
-      return kError;
-    }
-
-    const std::string public_key = "/etc/cvmfs/keys/" + repo_name + ".pub";
-    if (!server_tool->InitVerifyingSignatureManager(public_key)) {
-      LogCvmfs(kLogReceiver, kLogSyslogErr,
-               "CommitProcessor - error: Could not initialize the signature "
-               "manager");
-      return kError;
-    }
-
-    shash::Any manifest_base_hash;
-    UniquePtr<manifest::Manifest> manifest(server_tool->FetchRemoteManifest(
-        params.stratum0, repo_name, manifest_base_hash));
-
-    LogCvmfs(kLogReceiver, kLogSyslog,
-             "CommitProcessor - lease_path: %s, new root hash: %s",
-             lease_path.c_str(),
-             manifest->catalog_hash().ToString(false).c_str());
-  }
+  const UniquePtr<manifest::Manifest> manifest_new(
+    manifest::Manifest::LoadFile(new_manifest_path));
```

Comment:
At this point the manifest file is unfortunately no longer around, this is a tmpfile essentially. An easy fix is to just get the hash directly from the merge tool. I have a patch I can add directly here if you agree @jblomer